corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -27,37 +28,73 @@ hqDefine("data_dictionary/js/data_dictionary", [
         self.name = name || gettext("No Name");
         self.url = "#" + name;
         self.fhirResourceType = ko.observable(fhirResourceType);
-        self.properties = ko.observableArray();
+        self.groups = ko.observableArray();
+
+        self.init = function (groupData, changeSaveButton) {
+            for (let group of groupData) {
+                let groupObj = groupsViewModel(group.id, group.name, group.description, group.index, self.name, group.deprecated);
+                if (group.id) {
+                    groupObj.name.subscribe(changeSaveButton);
+                    groupObj.description.subscribe(changeSaveButton);
+                    groupObj.deprecated.subscribe(changeSaveButton);
```

Comment:
we don't need to subscribe for index changes? or that is handled elsewhere?
also, why the check for group.id before?

Reply:
The index changes are directly taken from the array of groups/properties itself, so no need to subscribe to index changes. Taking index directly from the array also reduces the additional work to keep the index in sync on group/properties. I added the group.id check for "No group" section.

Reply:
Thanks

> I added the group.id check for "No group" section.

Please add a comment in code. The special handling for "No group", does mud things around a bit.

Reply:
This check was not needed i have removed it. "No Group" section uses the same view model as other groups so the `group_obj.name.subscribe` wont show undefined anymore. [a200564](https://github.com/dimagi/commcare-hq/pull/32878/commits/a200564f7de41d3dc68d92ac8d807ee67a91bf0c)